src/domain.rs

```diff
@@ -151,29 +152,46 @@ pub struct OpenAIFileResp {
 
 impl OpenAIFile {
     //Constructor
-    pub async fn new(file_bytes: Vec<u8>, open_ai_key: &str, debug: bool) -> Result<Self> {
+    pub async fn new(
+        file_name: &str,
+        file_bytes: Vec<u8>,
+        open_ai_key: &str,
+        debug: bool,
+    ) -> Result<Self> {
```

Comment:
@darioalessandro, this is a breaking change the requires to pass the file name in the constructor. Would love your opinion on this.